tests/js/legendTests.js

```diff
@@ -337,6 +367,37 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
 
     });
 
+    jqUnit.test("Test the legend component with custom value and label display templates", function () {
+        jqUnit.expect(4);
+
+        var that = floe.tests.chartAuthoring.pieChart.legend(".floe-ca-legend-custom-labels", {
+            model: {
+                dataSet: floe.tests.chartAuthoring.objectArrayForCustomDisplay
+            },
+            legendOptions: {
+                sort:false,
+                colors: null,
+                labelTextDisplayTemplate: "%label (%percentage%)",
+                valueTextDisplayTemplate: "%value (out of %total)"
+            }
+        });
+
+        var d3LabelCells = that.jQueryToD3($(that.locate("labelCell")));
+        d3LabelCells.each(function (d,i) {
+            var expectedLabel = floe.tests.chartAuthoring.expectedCustomDisplay[i].expectedLabel;
+            jqUnit.assertEquals("The data labels are applied correctly in the legend", expectedLabel, ($(this).html()));
+        });
+
+        var d3ValueCells = that.jQueryToD3($(that.locate("valueCell")));
```

Comment:
Line 385-389 and line 391-397 perform similar action, can you create a function to share?